### PR TITLE
#CP-23 Fix CORS error in credential request

### PR DESCRIPTION
**Pull request by feyton**
### What does this PR do?
Fix the issue of the server not accepting `Cross Origin  Request with credentials flag`. This issue is a result of using **`cors()`** without a list of allowed origins which will reject all request that request credentials like **`cookie`** that is **http-only** or has **`secure:true`** flag.

### Description of Task to be completed

- [x] Requests with credentials= true flag are accepted
- [x] Documentation site using swagger is working
- [x] All tests are corrected to reflect changes
- [x] It is possible to send the refresh token request from the front-end

### How should this be manually tested?
The tests were not affected. This can be checked using the front-end part related to login where you can 
```
git clone https://github.com/atlp-rwanda/phantom-be-codebandits
cd phantom-be-codebandits
git fetch origin
git checkout bg-cors-whitelist-cp-23
npm install
//configure the .env according to the .env.sample file
// start the database
npm run dev
// open browser or postman (Check linked public workplace for testing)
```
This same code has proven to be working on hosted version here [https://phantom-codebandits.herokuapp.com](https://phantom-codebandits.herokuapp.com)
### Any background context you want to provide?

When you send a request that have a credentials while the `Allowed Origin="*"` which is the default when you just use `app.use(cors())` the request is denied by the server for security reasons. This PR fixes that by using the custom function to process the origin

### What are the relevant pivotal tracker stories?
["#CP-23"](https://trello.com/c/e9bo8lA8)

## Relevant screenshots
image

## The process and stages in line with the PR.
1. A user send a request with expired token
image
2. The request is intercepted and using the secure cookie it send a refresh token 
image
3. The request is being retried once after attaching the updated token
image

### CODE REVIEW

File: src/controllers/refreshToken.controller.js [hunk 1]
```diff
@@ -29,9 +29,9 @@ const refresh = async (req, res) => {
 		const accessToken = jwt.sign(
 			{ id: userToken.user.id, role: userToken.user.role },
 			process.env.ACCESS_TOKEN_SECRET,
-			{ expiresIn: '600s' }
+			{ expiresIn: '120s' }
```
Comment (renepromesse) [inline src/controllers/refreshToken.controller.js:32]: why did you change the expiration milliseconds?

**Review by renepromesse — COMMENTED** [review]
Hey @feyton , this is good. ignore the feedback if you had a reason to change it

File: src/controllers/refreshToken.controller.js (hunk 1 above)
Comment (feyton) [reply src/controllers/refreshToken.controller.js:32]: We were using that value to tackle the change of timeout while we were working on the integration since we had no refresh token access. I changed it because merging this PR will fix the issue and all requests can be intercepted automatically

**Review by samyshyaka — DISMISSED** [review]
Nice work @feyton, I was having Cross origin errors when I was trying to implement the integration of the logout feature to the frontend. This will definitely solve the issue.

**Review by Yvad60 — DISMISSED** [review]
Nice work @feyton  this one looks good to me Approved!

File: src/controllers/auth.controller.js [hunk 2]
```diff
@@ -17,14 +17,16 @@ const AuthHandler = async (req, res) => {
 			const accessToken = jwt.sign(
 				{ id: user.id, role: user.role },
 				process.env.ACCESS_TOKEN_SECRET,
-				{ expiresIn: '600s' }
+				{ expiresIn: '120s' }
```
Comment (feyton) [inline src/controllers/auth.controller.js:20]: The duration has been reduced since on the front-end all requests will be intercepted to fetch a new refresh token

File: src/controllers/refreshToken.controller.js [hunk 3]
```diff
@@ -29,9 +29,9 @@ const refresh = async (req, res) => {
...
 		);
-		return handleResponse(res, 200, { token: accessToken });
+		return handleResponse(res, 200, { access_token: accessToken });
```
Comment (feyton) [inline src/controllers/refreshToken.controller.js:34]: This line has been changed to be consistent with the **auth** route which return `access_token` in `data` too

**Review by feyton — COMMENTED** [review]
## Review information

**Review by Yvad60 — DISMISSED** [review]
Approved!

**Review by Yvad60 — APPROVED** [review]
Approved @feyton

**Review by renepromesse — APPROVED** [review]
